extensions/amp-apester-media/0.1/amp-apester-media.js

```diff
@@ -151,11 +156,12 @@ class AmpApesterMedia extends AMP.BaseElement {
       renderer: true,
       tags: extractTags(this.getAmpDoc(), this.element),
     };
+    observe(this.element, (inViewport) => this.viewportCallback_(inViewport));
   }
 
   /** @override */
   firstLayoutCompleted() {
-    this.viewportCallback(this.isInViewport());
+    this.viewportCallback_(this.isInViewport());
```

Comment:
We should probably no longer rely on `this.isInViewport()`? Also, maybe we should just do the same thing here that you did for `amp-anim` (with the same caveat that I commented on).

Reply:
definitely. `this.isInVieport_` needs to be removed since it is dependent on the same pathway in resources

Reply:
making this change now, and spinning out all "lower risk" components from ads

Reply:
actually, looks like there is a specific path to account for when this component is made visible but hasn't completed layout yet: (inViewport &&  !this.ready_) --> showPlaceholder

Reply:
Hmm. I think this should simply be: before layout complete show placeholder, after layout complete don't show placeholder.

Reply:
Reading into the historical context, and I'm still not sure why this component has custom placeholder info: https://github.com/ampproject/amphtml/pull/4291

Reply:
Not sure. But otherwise, the original rules seem pretty simple. Placeholder is hidden right after layout.